pulp_2_tests/tests/rpm/api_v2/test_iso_sync_publish.py

```diff
@@ -171,3 +173,31 @@ def test_all(self):  # pylint:disable=no-self-use
         path = urljoin(urljoin('/pulp/isos/', repo['id'] + '/'), iso_name)
         iso2 = client.get(path).content
         self.assertEqual(iso, iso2)
+
+
+class SyncInvalidFileTestCase(unittest.TestCase):
+    """Perform sync on an invalid files repository.
+
+    This test trgets the following issues:
+
+    * `Pulp 2 Tests #18 <https://github.com/PulpQE/Pulp-2-Tests/issues/18>`_.
+    * `Pulp 3845# <https://pulp.plan.io/issues/3845>`_.
+    """
+
+    def test_all(self):
+        """Perform sync on an invalid file repository."""
+        cfg = config.get_config()
+        client = api.Client(cfg, api.json_handler)
+        body = {
+            'id': utils.uuid4(),
+            'importer_config': {'feed': FILE_INVALID_FEED_URL},
+            'importer_type_id': 'iso_importer',
+        }
+        repo = client.post(REPOSITORY_PATH, body)
+        self.addCleanup(client.delete, repo['_href'])
+        with self.assertRaises(TaskReportError) as context:
+            sync_repo(cfg, repo)
+        for key in ('1048576 bytes', '5 bytes'):
```

Comment:
is this numbers constant? by any chance it can be updated on fixtures? shouldn't be better to have it dynamically loaded?

Reply:
@rochacbruno, those values come from here. [fixtures](https://github.com/PulpQE/pulp-fixtures/blob/512fe38443680a25eb937d3adc2ef87456c3f4fa/Makefile#L186)
In the test description was suggested to make assertions about the whole error message, but I decide to focus on the reason of failure. If you think a different kind of assertion should be applied, I can change. Suggestions?